tensorflow/core/kernels/one_hot_op.cc

```diff
@@ -113,45 +113,77 @@ class OneHotOp : public OpKernel {
   TF_DISALLOW_COPY_AND_ASSIGN(OneHotOp);
 };
 
-#define REGISTER_ONE_HOT(type)                           \
+#define REGISTER_ONE_HOT_32(type)                        \
   REGISTER_KERNEL_BUILDER(Name("OneHot")                 \
                               .Device(DEVICE_CPU)        \
+                              .TypeConstraint<int32>("I")\
                               .TypeConstraint<type>("T") \
                               .HostMemory("depth"),      \
-                          OneHotOp<CPUDevice, type>);
+                          OneHotOp<CPUDevice, type, int32>);
 
-TF_CALL_NUMBER_TYPES(REGISTER_ONE_HOT);
+#define REGISTER_ONE_HOT_64(type)                        \
+  REGISTER_KERNEL_BUILDER(Name("OneHot")                 \
+                              .Device(DEVICE_CPU)        \
+                              .TypeConstraint<int64>("I")\
+                              .TypeConstraint<type>("T") \
+                              .HostMemory("depth"),      \
+                          OneHotOp<CPUDevice, type, int64>);
+
+TF_CALL_NUMBER_TYPES(REGISTER_ONE_HOT_32);
```

Comment:
I had similar thoughts, but didn't want to change it for this first pass. I'll try to get strings working with the next push.